MediaBrowser.Common/Providers/ProviderIdParsers.cs

```diff
@@ -0,0 +1,125 @@
+﻿#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace MediaBrowser.Common.Providers
+{
+    /// <summary>
+    /// Parsers for provider ids.
+    /// </summary>
+    public static class ProviderIdParsers
+    {
+        private const int ImdbMinNumbers = 7;
+        private const int ImdbMaxNumbers = 8;
+
+        /// <summary>
+        /// Parses an IMDb id from a string.
+        /// </summary>
+        /// <param name="text">The text to parse.</param>
+        /// <param name="imdbId">The parsed IMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseImdbId(ReadOnlySpan<char> text, [NotNullWhen(true)] out string? imdbId)
+        {
+            var tt = "tt".AsSpan();
+
+            // imdb id is at least 9 chars (tt + 7 numbers)
+            while (text.Length >= 2 + ImdbMinNumbers)
+            {
+                var ttPos = text.IndexOf(tt);
+                if (ttPos == -1)
+                {
+                    imdbId = default;
+                    return false;
+                }
+
+                text = text.Slice(ttPos + tt.Length);
+                var i = 0;
+                for (; i < Math.Min(text.Length, ImdbMaxNumbers); i++)
```

Comment:
`Math.Min(text.Length, ImdbMaxNumbers)` should be calculated outside of the loop